src/main/java/com/ibm/stocator/fs/ObjectStoreFileSystem.java

```diff
@@ -287,16 +288,16 @@ public boolean delete(Path f, boolean recursive) throws IOException {
   @Override
   public FileStatus[] listStatus(Path f) throws FileNotFoundException, IOException {
     LOG.debug("List status of {}", f.toString());
-    return listStatus(f, null);
+    return listStatus(f, DEFAULT_FILTER);
```

Comment:
@djalova I wonder, if it's DEFAULT_FILTER, then it basically does nothing and return the same objects as was the input. I wonder, if it's DEFAULT_FILTER, can we avoid additional pass and just return list as is? My point is, why we need this for DEFAULT_FILTER:

for (FileStatus fs : listing) {
 +        if (filter.accept(fs.getPath())) {
 +          result.add(fs);
 +        }
 +      }